classes/Autocomplete.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace VUMC\REDCapInstrumentAndVariableSQLRenamer;
+
+class Autocomplete
+{
+    public static function replaceTerm($match)
+    {
+        $applyTag = function ($found) {
+            // the sorrounding tag can be customized here
+            $tagged = sprintf('<mark style="padding:0;background-color:yellow;""><b>%s</b></mark>', $found);
+            return $tagged;
+        };
+        $found = @$match[0];
+        if (!$found) {
+            return '';
+        }
+        return $applyTag($found);
+    }
+
+    public static function getTermRegExp($terms)
+    {
+        $termsReducer = function ($carry, $term) {
+            $quotedTerm = preg_quote($term); // we do not want to use regexps provided by the user interface
+            $normalized = "($quotedTerm)"; // enclose in grouping parenthesis
+            $carry[] = $normalized;
+            return $carry;
+        };
+        $result = array_reduce($terms, $termsReducer, []);
+        $regExp = sprintf('/%s/i', implode('|', $result));
+        return $regExp;
+    }
+
+    public static function searchTerms($terms, $text)
+    {
+        $regExp = self::getTermRegExp($terms);
+        $result = preg_replace_callback($regExp, 'self::replaceTerm', $text);
+        return $result;
+    }
+
+    public static function getAutocompleteData($module, $pid, $getTerm, $type, $option)
```

Comment:
high level items to consider:
* breaking this  method into three modular methods. 
* one for sanatizing / preparing the terms for the query
* another for running the query and returning the results, 
* and a third method for processing the results into a format to be consumed by the front end. 
* consider using data classes to represent each return result, with the properties: value, label, info group
* consider making `$users` a property on this class to be hydrated / populated by the various methods